src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -2261,6 +2266,147 @@ test_bypass_spawning_via_bypassAutoEncryption (void *unused)
    bson_destroy (kms_providers);
 }
 
+
+static mongoc_client_encryption_t *
+_make_kms_certificate_client_encryption (mongoc_client_t *client,
+                                         bson_error_t *error)
+{
+   mongoc_client_encryption_opts_t *client_encryption_opts =
+      mongoc_client_encryption_opts_new ();
+
+   {
+      bson_t *kms_providers = _make_aws_kms_provider (NULL);
+      mongoc_client_encryption_opts_set_kms_providers (client_encryption_opts,
+                                                       kms_providers);
+      bson_destroy (kms_providers);
+   }
+
+   mongoc_client_encryption_opts_set_keyvault_namespace (
+      client_encryption_opts, "keyvault", "datakeys");
+   mongoc_client_encryption_opts_set_keyvault_client (client_encryption_opts,
+                                                      client);
+
+   mongoc_client_encryption_t *client_encryption =
+      mongoc_client_encryption_new (client_encryption_opts, error);
+   ASSERT_OR_PRINT (client_encryption, (*error));
+
+   mongoc_client_encryption_opts_destroy (client_encryption_opts);
+   return client_encryption;
+}
+
+
+static void
+test_kms_tls_cert_valid (void *unused)
+{
+   bson_error_t error;
+   const int32_t connecttimeoutms = MONGOC_DEFAULT_CONNECTTIMEOUTMS;
+   const int is_client = 1;
+
+   mongoc_host_list_t host;
+   ASSERT_OR_PRINT (
+      _mongoc_host_list_from_string_with_err (&host, "127.0.0.1:7999", &error),
```

Comment:
`test_kms_tls_cert_valid`, `test_kms_tls_cert_expired`, and `test_kms_tls_cert_wrong_host` rely on three instances of the mock KMS server running, on ports 7999, 8000, and 8001.

I suggest noting that assumption in CONTRIBUTING.md. It contains notes about which environment variables are required to run CSFLE tests.